lxd/storage_pools.go

```diff
@@ -674,27 +674,41 @@ func storagePoolDeleteCheckPreconditions(cluster *db.Cluster, poolName string, p
 		return response.InternalError(err)
 	}
 
+	var projects []string
+
+	err = cluster.Transaction(func(tx *db.ClusterTx) error {
+		projects, err = tx.ProjectNames()
+		return err
+	})
+	if err != nil {
+		return response.InternalError(err)
+	}
+
 	if len(volumeNames) > 0 {
-		volumes, err := cluster.StoragePoolVolumesGet("default", poolID, supportedVolumeTypes)
-		if err != nil {
-			return response.InternalError(err)
-		}
+		for _, project := range projects {
+			volumes, err := cluster.StoragePoolVolumesGet(project, poolID, supportedVolumeTypes)
+			if err != nil {
+				return response.InternalError(err)
+			}
 
-		for _, volume := range volumes {
-			if volume.Type != "image" {
-				return response.BadRequest(fmt.Errorf("storage pool \"%s\" has volumes attached to it", poolName))
+			for _, volume := range volumes {
+				if volume.Type != "image" {
+					return response.BadRequest(fmt.Errorf("storage pool \"%s\" has volumes attached to it", poolName))
```

Comment:
@monstermunchkin you could use `%q` here rather than `\"%s\"`

Reply:
Fixed